spdx-utils/src/main/kotlin/SpdxExpression.kt

```diff
@@ -33,7 +33,7 @@ import org.antlr.v4.runtime.CommonTokenStream
  * [1]: https://spdx.dev/spdx-specification-21-web-version#h.jxpfx0ykyb60
  */
 @JsonSerialize(using = ToStringSerializer::class)
-sealed class SpdxExpression {
+sealed class SpdxExpression : Comparable<SpdxExpression> {
```

Comment:
Have you considered using comparator instead? It has the advantages of
1. Being more decoupled
2. Users of the sorting can be determined via the IDE easily and thus side effects of changes to the sorting order can be figured out
3. no need to worry about consistency with `equals()`

Reply:
The main point of this change was to provide a sensible default way to sort [SpdxExpression]s, *if* the user wants / needs sorting. It was triggered by [this proposed change](https://github.com/oss-review-toolkit/ort/pull/3566#discussion_r568395756) so you can stop writing something as clumsy as `.flatMapTo(sortedSetOf((compareBy { it.toString() })))` and just use `.flatMapTo(sortedSetOf())`.

If by "using comparator instead" you mean to *not* implement `Comparable<SpdxExpression>` but just provide a `val COMPARATOR = compareBy<SpdxExpression> { it.license.toString() }` for arbitrary use, then that would not really help to simplify the syntax in the above example. And if you mean to use a (private?) comparator *and* implement `Comparable<SpdxExpression>`, I do not really see an advantage to the current proposal.

Reply:
> If by "using comparator instead" you mean to *not* implement `Comparable<SpdxExpression>` but just provide a [...]

That's what I meant. Anyhow, won't insist on it.